x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -88,9 +91,11 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         boolean isWatcherStoppedManually = isWatcherStoppedManually(event.state());
+        boolean isStoppedOrStopping = this.state.get() == WatcherState.STOPPED ||
+            this.state.get() == WatcherState.STOPPING;
         // if this is not a data node, we need to start it ourselves possibly
         if (event.state().nodes().getLocalNode().isDataNode() == false &&
-            isWatcherStoppedManually == false && this.state.get() == WatcherState.STOPPED) {
+            isWatcherStoppedManually == false && isStoppedOrStopping) {
             this.state.set(WatcherState.STARTING);
             watcherService.start(event.state(), () -> this.state.set(WatcherState.STARTED));
```

Comment:
Should we guard against going into STARTED state from a state other than STARTING?
(`this.state.compareAndSet(WatcherState.STARTING, WatcherState.STARTED);`)

Reply:
I am trying to avoid changes to the `START*` states/behavior. This change _should_ only impact the `STOPPED` state for a manual (e.g. via the API) request to stop.